tracer/src/Datadog.Trace/HttpOverStreams/ChunkedEncoding/ChunkedEncodingWriteStream.cs

```diff
@@ -0,0 +1,117 @@
+﻿// <copyright file="ChunkedEncodingWriteStream.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#nullable enable
+
+using System.Globalization;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Datadog.Trace.HttpOverStreams.ChunkedEncoding;
+
+internal class ChunkedEncodingWriteStream(Stream innerStream) : DelegatingStream(innerStream)
+{
+    private static readonly byte[] CrLfBytes = { 0x0D, 0x0A };
+    private static readonly byte[] FinalChunkBytes = { 0x30, 0x0D, 0x0A, 0x0D, 0x0A, };
+
+    private readonly Stream _innerStream = innerStream;
+    private readonly byte[] _chunkSizeBuffer = new byte[8]; // max length of Int32 as UTF-8 hex
+
+    public override async Task WriteAsync(byte[] buffer, int offset, int count, CancellationToken cancellationToken)
+    {
+        // don't want to ever send a zero-length terminator unless we're at the end
+        if (count == 0)
+        {
+            // Don't write if nothing was given, especially since we don't want to accidentally send a 0 chunk,
+            // which would indicate end of body.  Instead, just ensure no content is stuck in the buffer.
+            await _innerStream.FlushAsync(cancellationToken).ConfigureAwait(false);
+            return;
+        }
+
+        // write the chunked encoding header
+        var bytesWritten = WriteChunkedEncodingHeaderToBuffer(_chunkSizeBuffer, count);
+        await _innerStream.WriteAsync(_chunkSizeBuffer, 0, bytesWritten, cancellationToken).ConfigureAwait(false);
+
+        // add the new line
+        await _innerStream.WriteAsync(CrLfBytes, offset: 0, count: 2, cancellationToken).ConfigureAwait(false);
+
+        // add the content
+        await _innerStream.WriteAsync(buffer, offset, count, cancellationToken).ConfigureAwait(false);
+
+        // add the extra new line
+        await _innerStream.WriteAsync(CrLfBytes, offset: 0, count: 2, cancellationToken).ConfigureAwait(false);
+    }
+
+    public override void Write(byte[] buffer, int offset, int count)
+    {
+        // don't want to ever send a zero-length terminator unless we're at the end
+        if (count == 0)
+        {
+            // Don't write if nothing was given, especially since we don't want to accidentally send a 0 chunk,
+            // which would indicate end of body.  Instead, just ensure no content is stuck in the buffer.
+            _innerStream.Flush();
+            return;
+        }
+
+        // write the chunked encoding header
+        var bytesWritten = WriteChunkedEncodingHeaderToBuffer(_chunkSizeBuffer, count);
+        _innerStream.Write(_chunkSizeBuffer, 0, bytesWritten);
+
+        // add the new line
+        _innerStream.WriteAsync(CrLfBytes, offset: 0, count: 2);
```

Comment:
You're actively trying to make the tests flaky, are you? 😛 

```suggestion
        _innerStream.Write(CrLfBytes, offset: 0, count: 2);
```